From 3c8bd4aaa3c871f5bfb064b740380edaaec23b92 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pierre-Cl=C3=A9ment=20Tosi?= <ptosi@google.com>
Date: Fri, 23 Sep 2022 08:10:15 +0000
Subject: [PATCH] [automerge] libfdt: fdt_path_offset_namelen: Reject empty
 paths 2p: a6ac6d916d 2p: c69e9730e6

Original change: https://googleplex-android-review.googlesource.com/c/platform/external/dtc/+/20024810

Bug: 246465319
Change-Id: I9096ca5ab6c194d2512e3c902a9658c33a40ff0c
(cherry picked from commit 5b78ca841071fee81479036462e7e048fefacb26)
Merged-In: I9096ca5ab6c194d2512e3c902a9658c33a40ff0c
---
 libfdt/fdt_ro.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libfdt/fdt_ro.c b/libfdt/fdt_ro.c
index e03570a..7a65dfe 100644
--- a/libfdt/fdt_ro.c
+++ b/libfdt/fdt_ro.c
@@ -253,6 +253,9 @@ int fdt_path_offset_namelen(const void *fdt, const char *path, int namelen)
 
 	FDT_RO_PROBE(fdt);
 
+	if (namelen < 1)
+		return -FDT_ERR_BADPATH;
+
 	/* see if we have an alias */
 	if (*path != '/') {
 		const char *q = memchr(path, '/', end - p);
-- 
2.38.1.273.g43a17bfeac-goog

